res/controllers/Denon-MC7000-scripts.js

```diff
@@ -0,0 +1,839 @@
+    		engine.setValue(group, "pitch_down", true);
+    	}
+    };
+    
+    // Assign Channel to Crossfader
+    MC7000.crossfaderAssign = function(channel, control, value, status, group) {
+	    // Centre position
+    	if (value === 0x00) {
+            engine.setValue(group, "orientation", 1);
+        }
+        // Left position
+	    else if (value === 0x01) {
+            engine.setValue(group, "orientation", 0);
+        }
+        // Right position
+	    else if (value === 0x02) {
+            engine.setValue(group, "orientation", 2);
+        }
+    };
+    
+    // Assign Spinback length to STOP TIME knob
+    MC7000.stopTime = function(channel, control, value, status, group) {
+    	var deckNumber = script.deckFromGroup(group);
+    	// "factor" for engine.brake()
+    	// this formula produces factors between 31 (min STOP TIME for ca 7 sec back in track) 
+    	// and 1 (max STOP TIME for ca 18.0 sec back in track)
+    	MC7000.factor[deckNumber] = (1.1 - (value / 127)) * 30 - 2;
+    };    
+    
+    // Use the CENSOR button as Spinback with STOP TIME adjusted length
+    MC7000.brake_button = function(channel, control, value, status, group) {
+        var deckNumber = script.deckFromGroup(group);
+    	var deck = parseInt(group.substring(8,9)); // work out which deck we are using 
+        engine.brake(deck, value > 0, MC7000.factor[deckNumber], - 15); // start at a rate of -15 and decrease by "factor"
+    };
+};
+
+/* SET CROSSFADER CURVE */
+MC7000.crossFaderCurve = function (control, value) {
+    script.crossfaderCurve(value);
+};
+
+/* Set FX wet/dry value */
+MC7000.fxWetDry = function(midichan, control, value, status, group) {
+	var numTicks = (value < 0x64) ? value: (value - 128);
+	var newVal = engine.getValue(group, "mix") + numTicks/64*2;
+	engine.setValue(group, "mix", Math.max(0, Math.min(1, newVal)));
+};
+
+/* Next Rate range calculation */
+MC7000.getNextRateRange = function(currRateRange) {
+    for (var i = 0; i < MC7000.posRateRanges.length; i++) {
+        if (MC7000.posRateRanges[i] > currRateRange) {
+            return MC7000.posRateRanges[i];
+        }
+    }
+    return MC7000.posRateRanges[0];
+};
+
+/* Previous Rate range calculation */
+MC7000.getPrevRateRange = function(currRateRange) {
+    for (var i = 0; i < MC7000.negRateRanges.length; i++) {
+        if (MC7000.negRateRanges[i] < currRateRange) {
+            return MC7000.negRateRanges[i];
+        }
+    }
+    return MC7000.negRateRanges[0];
+};
+        
+/* LEDs for VuMeter */
+// VuMeters only for Channel 1-4 / Master is on Hardware
+MC7000.VuMeter = function(value, group) {
+	var deckNumber = script.deckFromGroup(group),
+	    peak = 0x76, // where the red LED starts (clipping indicator)
+        level = value*value*value*value*0x69;
+
+    if (engine.getValue(group, "PeakIndicator")) {
+        var level = peak;
+    }
+    // now send the level meter signal to controller
+    midi.sendShortMsg(0xB0 + deckNumber - 1, 0x1F, level);
+};
```

Comment:
```suggestion
// where the red LED starts (clipping indicator)
var MC7000.VuMeterLEDPeakValue = 0x76

MC7000.VuMeter = function(value, group) {
  var VULevelOutValue = engine.getValue(group, "PeakIndicator") ? 
    MC7000.VuMeterLEDPeakValue : 
    value*value*value*value*0x69;

 	var deckNumber = script.deckFromGroup(group);

 	midi.sendShortMsg(0xB0 + deckNumber - 1, 0x1F, VULevelOutValue);
};
```

Reply:
looks much nicer, implemented this. Thanks a lot.

Reply:
Once you modified the code, it would make sense for you to push it as soon as possible so we can reiterate on it.

Reply:
> Readibility is subjective. The purpose of the guidelines is to make the codebase consistent. Can you commit and push your progress?
 I am sorry but a bit lost now as also this is first time on GitHub. Please can you advise? I used following commands:
`$ git add res/controllers/Denon-MC7000-scripts.js res/controllers/Denon-MC7000.midi.xml`
`$ git status` showed the 2 files ready marked
`$ git commit -m "Denon MC7000 mapping"`
`$ git push --set-upstream origin Denon_MC7000_mapping`

but I don't see them updated in my Denon_MC7000_mapping branch when looking on the website of github.com. Any suggestion is welcome

Reply:
Yes. thats what you did for you initial push. Now you need to add the mapping again, so `git add res/controllers/Denon-MC7000-scripts.js` (we can ignore the xml as we didn't make any changes to it). Then you do a `git commit -m "incorporated suggestions from first review"` (usually you should commit small complete chunks of changes; so for example when we changed the VuMeter function; but in this case its fine this way as well) and then you do a `git push origin Denon_MC7000_mapping` (we omit the `--set-upstream` because  we already set it the first time you pushed).

Reply:
And copy the output if you can so we can troubleshoot error messages.

Reply:
```
$ git status
Auf Branch Denon_MC7000_mapping
Ihr Branch ist auf demselben Stand wie 'origin/Denon_MC7000_mapping'.

zum Commit vorgemerkte Änderungen:
(benutzen Sie "git reset HEAD <Datei>..." zum Entfernen aus der Staging-Area)

geändert:       res/controllers/Denon-MC7000-scripts.js
geändert:       res/controllers/Denon-MC7000.midi.xml
```
```
$ git commit -m "incorporated suggestions from first review"
Check for byte-order marker..............................................Passed
Check for case conflicts.................................................Passed
Check JSON...........................................(no files to check)Skipped
Check for merge conflicts................................................Passed
Check Xml................................................................Passed
Check Yaml...........................................(no files to check)Skipped
Fix End of Files.........................................................Passed
Mixed line ending........................................................Passed
Trim Trailing Whitespace.................................................Passed
Don't commit to branch...................................................Passed
eslint...................................................................Passed
clang-format.............................................................Failed
- hook id: clang-format
- exit code: 1

git: 'clang-format' ist kein Git-Befehl. Siehe 'git --help'.

Der ähnlichste Befehl ist
clang-format-6.0

black................................................(no files to check)Skipped
flake8...............................................(no files to check)Skipped
qsscheck.............................................(no files to check)Skipped
```
```
$ git push origin Denon_MC7000_mapping
Everything up-to-date
```

Reply:
Seems like you are not commiting because the pre-commit hook fails. @Holzhaus Can you help troubleshoot this?